core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -556,6 +562,46 @@ public void copyLogSegmentsToRemote(UnifiedLog log) throws InterruptedException
             }
         }
 
+        public void cleanupDeletedRemoteLogSegments() {
+            if (isCancelled())
+                return;
+
+            Uuid topicId = topicIdPartition.topicId();
+            if (deletedTopicIds.contains(topicId)) {
+                cleanupAllRemoteLogSegments();
+                cancelRLMtask();
+                deletedTopicIds.remove(topicId);
+            }
+        }
+
+        private void cleanupAllRemoteLogSegments() {
+            if (!isLeader())
```

Comment:
Hmm, after a topic is deleted, there is no leader.

Reply:
This is used to verify whether the current RLMTask is a leader, and the leaderEpoch information stored in RLMTask will not change after the topic is deleted.

Reply:
This would still be a bit weird. It's possible for a partition to be already leaderless (for example, no replica in ISR) when a topic is deleted. How is that case handled?